src/core/ext/filters/client_channel/resolver/xds/xds_resolver.cc

```diff
@@ -469,13 +489,48 @@ ConfigSelector::CallConfig XdsResolver::XdsConfigSelector::GetCallConfig(
     XdsResolver* resolver =
         static_cast<XdsResolver*>(resolver_->Ref().release());
     ClusterState* cluster_state = it->second->Ref().release();
+    // Generate a hash
+    absl::optional<uint64_t> hash;
+    for (const auto& hash_policy : entry.route.hash_policies) {
+      absl::optional<uint64_t> new_hash;
+      uint64_t key;
+      switch (hash_policy.type) {
+        case XdsApi::Route::HashPolicy::HEADER:
+          new_hash = HeaderHashHelper(hash_policy, args.initial_metadata);
+          break;
+        case XdsApi::Route::HashPolicy::CHANNEL_ID:
+          key = static_cast<uint64_t>(reinterpret_cast<uintptr_t>(resolver));
```

Comment:
Not sure if this applies to C-core. In Java, the resolved will be shut down when the Channel enters IDLE and a new resolver will be created after the Channel exists IDLE mode. If you hash based on the resolver address, RPCs may get different hashes even if they are sent through the same Channel. We have to use the _real unique Channel identity_.

Reply:
Yeah, the same is true in C-core: if we go idle, we will destroy the resolver and create a new one later when we leave IDLE.  And yes, this means that the hash will change whenever this happens.  This is not ideal in principle, but in practice I think it's going to be fine, since going IDLE will always stop using the subchannels anyway, and we'll need to establish new sessions when we leave IDLE.  And this approach was easier than trying to plumb the address of the channel down into the resolver.